fontforgeexe/wordlistparser.c

```diff
@@ -351,6 +361,177 @@ unichar_t* WordlistEscapedInputStringToRealStringBasic(
     return ret;
 }
 
+/************************************************************/
+/************************************************************/
+/************************************************************/
+
+/**
+ * This is the array of characters that are referenced from a
+ * wordlistline. I'd felt that such a setup was much better than the
+ * previous string wrangling, and with fonts that have splinechars
+ * that have no unicode value, it makes more sense to explicitly
+ * address the string this way.
+ * 
+ * for example,
+ * ab/comma/slash will have 4 elements in the array and a null sc as element this[4].
+ * this[0].sc = 'a'
+ * this[1].sc = 'b'
+ * this[2].sc = ','
+ * this[3].sc = '/'
+ * this[4].sc = \0
+ *
+ * currentGlyphIndex is a hang over from old code, still might be handy if you have a pointer
+ * to a single WordListChar element and you want to know what splinechar it is in the string.
+ * Selections are now handled using isSelected in each element.
+ */
+typedef struct wordlistchar {
+    SplineChar* sc;
+    int isSelected;
+    int currentGlyphIndex;
+} WordListChar;
+
+static int WordListLineSz = 1024;
+typedef WordListChar* WordListLine;
+
+int WordListLine_countSelected( WordListLine wll )
+{
+    int ret = 0;
+    for( ; wll->sc; wll++ ) {
+	ret += wll->isSelected;
+    }
+    return ret;
+}
+
+
+WordListLine WordlistEscapedInputStringToParsedDataComplex(
+    SplineFont* sf,
+    const unichar_t* input_const,
+//    GArray** selected_out,
+    WordlistEscapedInputStringToRealString_getFakeUnicodeOfScFunc getUnicodeFunc,
+    void* udata )
+{
+    char* input = u2utf8_copy(input_const);
+
+    // truncate insanely long lines rather than crash
+    if( strlen(input) > PATH_MAX )
+	input[PATH_MAX] = '\0';
+
+    WordListChar* ret = calloc( WordListLineSz, sizeof(WordListChar));
+    WordListChar* out = ret;
+    char* in = input;
+    char* in_end = input + strlen(input);
+    // trim comment and beyond from input
```

Comment:
@monkeyiq Why would there be comments in the input field? If `# sth` an accepted syntax for comments in a wordlist?  
Asking this because otherwise we could stop forbidding \# in its raw form.